### PR TITLE
Added lstm stock prediction

### PR DESCRIPTION
**Pull request by Tejaswgupta**
### **Describe your change:**

Pr against issue #1770 

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: neural_network/lstm_stock/core/model.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	def load_model(self, filepath):
+		print('[Model] Loading model from file %s' % filepath)
+		self.model = load_model(filepath)
+
+	def build_model(self, configs):
+		timer = Timer()
+		timer.start()
+
+		for layer in configs['model']['layers']:
+			neurons = layer['neurons'] if 'neurons' in layer else None
```
Comment (cclauss) [inline neural_network/lstm_stock/core/model.py]: Let's use dict.get()... https://docs.python.org/3/library/stdtypes.html#dict.get

`neurons = layer.get('neurons')`  # Repeat below...

File: neural_network/lstm_stock/run.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        steps_per_epoch=steps_per_epoch,
+        save_dir=configs['model']['save_dir']
+    )
+
+    x_test, y_test = data.get_test_data(
+        seq_len=configs['data']['sequence_length'],
+        normalise=configs['data']['normalise']
+    )
+
+    predictions = model.predict_sequences_multiple(x_test, configs['data']['sequence_length'], configs['data']['sequence_length'])
```
Comment (cclauss) [inline neural_network/lstm_stock/run.py]: Please format code with psf/black to limit lines to 88 characters max.

File: neural_network/lstm_stock/core/data_processor.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
+import math
+import numpy as np
+import pandas as pd
+
+class DataLoader():
+    """A class for loading and transforming data for the lstm model"""
+
+    def __init__(self, filename, split, cols):
```
Comment (cclauss) [inline neural_network/lstm_stock/core/data_processor.py]: Type hint and doctests please.

File: neural_network/lstm_stock/core/data_processor.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        Create x, y test data windows
+        Warning: batch method, not generative, make sure you have enough memory to
+        load data, otherwise reduce size of the training split.
+        '''
+        data_windows = []
+        for i in range(self.len_test - seq_len):
+            data_windows.append(self.data_test[i:i+seq_len])
+
+        data_windows = np.array(data_windows).astype(float)
+        data_windows = self.normalise_windows(data_windows, single_window=False) if normalise else data_windows
```
Comment (cclauss) [inline neural_network/lstm_stock/core/data_processor.py]: This line is too long for a ternary if...  Just do a normal if and wrap lines to 88 char max.

**Comment by Tejaswgupta** [comment]
So should I fix or it doesn't belong here?

**Comment by onlinejudge95** [comment]
@Tejaswgupta let's do a spike. How about you abstract away all the `configurations` out and only submit the algorithm. That being said also make sure to document your configuration if required in module level docstring. I guess then we will have enough clarity to decide on the same.

PS- In any case it would be good if you address the changes.

**Comment by cclauss** [comment]
https://travis-ci.com/TheAlgorithms/Python/builds/149870316#L804

**Pull request closed without merging**